src/shared/clients/s3Client.ts

```diff
@@ -330,13 +330,24 @@ export class DefaultS3Client {
             })
         }
 
-        try {
-            await Promise.all([promisifyReadStream(readStream), managedUploaded.promise()])
-        } catch (e) {
-            getLogger().error('Failed to upload %s to bucket %s: %O', request.key, request.bucketName, e)
-            throw e
-        }
-        getLogger().debug('UploadFile succeeded')
+        Promise.all([promisifyReadStream(readStream), managedUploaded.promise()]).then(
+            () => {
+                getLogger().debug('UploadFile succeeded')
+            },
+            err => {
+                getLogger().error('Failed to upload %s to bucket %s: %O', request.key, request.bucketName, err)
+                throw err
+            }
+        )
```

Comment:
I don't think this is the right syntax. I think you'll want something like 
```
.then( (val, err) => {
    if (err) {
        getLogger().error('Failed to upload %s to bucket %s: %O', request.key, request.bucketName, err)
        throw err
    }
    getLogger().debug('UploadFile succeeded')
}

Reply:
It gives an error if I do it like that